File: Data Structures & Algorithms/permutations-ii/submission-5.py
class Solution:
    def permuteUnique(self, nums: List[int]) -> List[List[int]]:
        res = []
        tmp = []

        cnt = {}
        for n in nums:
            cnt[n] = cnt.get(n,0)+1
        
        def f():
            if len(tmp)==len(nums):
                res.append(tmp[:])
                return
            
            for n in cnt:
                if cnt[n]>0:
                    cnt[n] -= 1
                    tmp.append(n)
                    f()
                    tmp.pop()
                    cnt[n] += 1
        
        f()
        return res

File: Data Structures & Algorithms/reverse-string/submission-5.cs
public class Solution {
    public void ReverseString(char[] s) {
        char[] tmp = new char[s.Length];
        int n = s.Length;

        for (int i=0; i<n; i++) {
            tmp[i] = s[n-1-i];
        }

        for (int i=0; i<n; i++) {
            s[i] = tmp[i];
        }

    }
}